ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -702,59 +696,69 @@ void writeIndexSegmentToFile(Offset safeEndPoint) throws IOException, StoreExcep
    */
   void seal() throws IOException, StoreException {
     sealed.set(true);
-    if (!config.storeKeepIndexInMemory) {
-      mmap();
-    }
+    map();
     // we should be fine reading bloom filter here without synchronization as the index is read only
     persistBloomFilter();
   }
 
   /**
-   * Memory maps the segment of index.
+   * Maps the segment of index either as a memory map or a in memory buffer depending on config.
    * @throws IOException if there is any problem reading or mapping files
    * @throws StoreException if there are problems with the index
    */
-  private void mmap() throws IOException, StoreException {
+  private void map() throws IOException, StoreException {
```

Comment:
Looks like `mapped` is set to true only when `sealed` is set to true. Can we just use one variable?

Reply:
That's right, they are the same with this change